File: mvbackteste/src/main/java/br/com/testeMV/model/Profissional.java
package br.com.testeMV.model;

import br.com.testeMV.model.dto.ProfissionalDTO;
import java.util.HashSet;
import java.util.Set;
import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table
@Builder
public class Profissional {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "profissional_id")
	private Long id;

	private String nome;
	private String telefone;
	private String cargo;

	@Embedded
	private Endereco endereco;


	@ManyToMany
	@JoinTable(name = "profissional_hospital",
			joinColumns = {@JoinColumn(name = "profissional_id")},
			inverseJoinColumns = {@JoinColumn(name = "hospital_id")})
	public Set<Hospital> hospitais = new HashSet<>();

	public ProfissionalDTO toDTO() {
		return ProfissionalDTO.builder()
				.id(id)
				.nome(nome)
				.telefone(telefone)
				.cargo(cargo)
				.hospitais(hospitais)
				.endereco(endereco)
				.build();
	}
}

File: frontmv2/src/app/model/profissional.ts

import { ProfissionalDTO } from './dto/profissionalDTO';
import { Endereco } from './endereco';
import { Hospital } from './hospital';

export class Profissional {
  id: number
  nome: string
  telefone: string
  cargo: string
  endereco: Endereco

  hospitais: Hospital[]

  static withId(id: number): Profissional {
    const profissional = new Profissional();
    profissional.id = id
    return profissional
  }
  static fromDTO(dto: ProfissionalDTO): Profissional {
    let profissional = new Profissional()
    profissional.id = dto.id
    profissional.nome = dto.nome
    profissional.telefone = dto.telefone
    profissional.hospitais = dto.hospitais
    profissional.cargo = dto.cargo
    let endereco = new Endereco();
    endereco.rua = dto.rua
    endereco.numero = dto.numero
    endereco.bairro = dto.bairro
    endereco.cidade = dto.cidade
    endereco.estado = dto.estado
    profissional.endereco = endereco
    return profissional
  }


  toDTO(): ProfissionalDTO {
    let dto = new ProfissionalDTO()
    dto.id = this.id
    dto.nome = this.nome
    dto.telefone = this.telefone
    dto.cargo = this.cargo
    dto.hospitais = this.hospitais
    dto.rua = this.endereco.rua
    dto.numero = this.endereco.numero
    dto.bairro = this.endereco.bairro
    dto.cidade = this.endereco.cidade
    dto.estado = this.endereco.estado

    return dto
  }
}